src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationNodes.vb

```diff
@@ -0,0 +1,1662 @@
+            MyBase.New(semanticModel, syntax, type, constantValue, isImplicit)
+            _operationFactory = operationFactory
+            _operand = operand
+        End Sub
+
+        Protected Overrides Function CreateOperand() As IOperation
+            Return SetParentOperation(_operationFactory.Create(_operand), Me)
+        End Function
+    End Class
+
+    Friend NotInheritable Class VisualBasicLazyPropertyInitializerOperation
+        Inherits LazyPropertyInitializerOperation
+
+        Private ReadOnly _operationFactory As VisualBasicOperationFactory
+        Private ReadOnly _value As BoundNode
+
+        Friend Sub New(operationFactory As VisualBasicOperationFactory, value As BoundNode, locals As ImmutableArray(Of ILocalSymbol), initializedProperties As ImmutableArray(Of IPropertySymbol), kind As OperationKind, semanticModel As SemanticModel, syntax As SyntaxNode, type As ITypeSymbol, constantValue As [Optional](Of Object), isImplicit As Boolean)
+            MyBase.New(locals, initializedProperties, kind, semanticModel, syntax, type, constantValue, isImplicit)
+            _operationFactory = operationFactory
+            _value = value
+        End Sub
+
+        Protected Overrides Function CreateValue() As IOperation
+            Return SetParentOperation(_operationFactory.Create(_value), Me)
+        End Function
+    End Class
+
+    Friend NotInheritable Class VisualBasicLazyPropertyReferenceOperation
+        Inherits LazyPropertyReferenceOperation
+
+        Private ReadOnly _operationFactory As VisualBasicOperationFactory
+        Private ReadOnly _instance As BoundNode
+        Private ReadOnly _boundProperty As BoundPropertyAccess
+
+        Friend Sub New(operationFactory As VisualBasicOperationFactory, instance As BoundNode, boundProperty As BoundPropertyAccess, [property] As IPropertySymbol, semanticModel As SemanticModel, syntax As SyntaxNode, type As ITypeSymbol, constantValue As [Optional](Of Object), isImplicit As Boolean)
+            MyBase.New([property], semanticModel, syntax, type, constantValue, isImplicit)
+            _operationFactory = operationFactory
+            _instance = instance
+            _boundProperty = boundProperty
+        End Sub
+
+        Protected Overrides Function CreateInstance() As IOperation
+            Return SetParentOperation(_operationFactory.CreateReceiverOperation(_instance, [Property]), Me)
+        End Function
+
+        Protected Overrides Function CreateArguments() As ImmutableArray(Of IArgumentOperation)
+            Return SetParentOperation(_operationFactory.DeriveArguments(_boundProperty), Me)
+        End Function
+    End Class
+
+    Friend NotInheritable Class VisualBasicLazyRangeCaseClauseOperation
+        Inherits LazyRangeCaseClauseOperation
+
+        Private ReadOnly _operationFactory As VisualBasicOperationFactory
+        Private ReadOnly _minimumValue As BoundNode
+        Private ReadOnly _maximumValue As BoundNode
```

Comment:
>Private ReadOnly _maximumValue As BoundNode [](start = 8, length = 43)

It looks like we can reduce the size of this node by storing a single BoundRangeCaseClause node instead of two of its children. Applies to other similar situations.

Reply:
My intent here was to try and keep as much of the knowledge about how to go from BoundNode->IOperation in the Factory as possible. Generally, I could have always taken the approach that we only hold onto the node that created this IOperation, but I felt that was leaking too much info about node creation back to the node implementation. The only times I break this is when, for child creation, the factory needed to have the original node. It does make the node classes slightly larger, but it was worth it in my opinion.

---
In reply to: [232116761](https://github.com/dotnet/roslyn/pull/31010#discussion_r232116761) [](ancestors = 232116761)